src/App.tsx

```diff
@@ -1,34 +1,89 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filteredBy, setFilteredBy] = useState('');
+
+  const [loading, setLoading] = useState(false);
+
+  const getFilteredTodos = (queryData: string, fieldFilter: string): Todo[] => {
+    let preparedTodos = [...todos];
+
+    if (queryData) {
+      preparedTodos = preparedTodos.filter(todo =>
+        todo.title.toLowerCase().includes(query.trim().toLowerCase()),
+      );
+    }
+
+    if (fieldFilter !== 'all') {
+      switch (filteredBy) {
+        case 'active':
+          preparedTodos = preparedTodos.filter(todo => !todo.completed);
+          break;
+        case 'completed':
+          preparedTodos = preparedTodos.filter(todo => todo.completed);
+          break;
+        default:
+          break;
```

Comment:
Never add `break` operator to the last case (not necessarily `case` , `default` cases as well) you're looking through. But currently you don't need to use `default` construction here. You can just omit this